framework/src/utils/LinearInterpolation.C

```diff
@@ -70,10 +87,22 @@ T
 LinearInterpolationTempl<T>::sampleDerivative(const T & x) const
 {
   // endpoint cases
-  if (x < _x[0])
-    return 0.0;
-  if (x >= _x[_x.size() - 1])
-    return 0.0;
+  if (_extrap)
+  {
+    if (x <= _x[0])
+      return (_y[1] - _y[0]) / (_x[1] - _x[0]);
+    if (x >= _x.back())
+    {
+      return (_y[_x.size() - 2] - _y.back()) / (_x[_x.size() - 2] - _x.back());
```

Comment:
shouldn't the first one technically be a `_y.size()` ?

Reply:
well, they should be the same

Reply:
Good catch! Should be _y.size(). Copied pasted wrong one.

Reply:
Yes, they are the same, but I fix it. Thanks.